backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -505,9 +485,17 @@ def manage_access_point_and_policy(self):
                 self.s3_prefix,
                 perms_to_actions(self.share.permissions, SidType.BucketPolicy),
             )
-        s3_client.attach_access_point_policy(
-            access_point_name=self.access_point_name, policy=json.dumps(access_point_policy)
-        )
+
```

Comment:
why do we need this now and didn't need it previously?

Reply:
This is in case AP is not yet created. Before the retry part was in creation of of AP

Reply:
I would explicitly encapsulate the Retry mechanism in the `create_bucket_access_point` to ensure that all calls to it return when the bucket is created.

For example something like the below....
```python
    def create_bucket_access_point(self, bucket_name: str, access_point_name: str):
        try:
            self._client.create_access_point(AccountId=self._account_id, Name=access_point_name, Bucket=bucket_name)
        except self._client.exceptions.AccessPointAlreadyOwnedByYou:
            ...
        except Exception as e:
            log.exception(f'S3 bucket access point creation failed for location {bucket_name=}')
            raise e
        return Retrying(
            retry_on_exception=self._client.exceptions.NotFoundException,
            stop_max_attempt_number=5,
            wait_random_min=1000,
            wait_random_max=3000,
        ).call(self.get_bucket_access_point_arn, access_point_name)['AccessPointArn']
```

Also you must make sure that you wait enough time. Previous code was waiting for 300secs++, yours wait from 5 to 15 seconds.
Also I think that the wait random is not a good idea, and should use exponential back off.

Reply:
Won't it lead us to the same error? If get_access_point_arn will fail, we again receive the problem of Negative Cash?

Reply:
also, there is no `self._client.exceptions.AccessPointAlreadyOwnedByYou:` so, I have to check the string